tests/test_frontend_vgg.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+import torch
+from torch import nn
+
+from i6_models.parts.frontend import (
+    VGG4LayerActFrontendV1,
+    VGG4LayerActFrontendV1Config,
+)
+
+
+def test_conformer_vgg_layer_act_frontend_v1():
+    torch.manual_seed(42)
+
+    def get_output_shape(
+        batch,
+        time,
+        time_padding,
+        features,
+        conv1_channels,
+        conv2_channels,
+        conv3_channels,
+        conv4_channels,
+        pool1_red,
+        pool2_red,
+        in_dim,
+        out_dim,
+    ):
+        data_input = torch.randn(batch, time, features)
+        data_input = torch.cat((data_input, torch.zeros(batch, time_padding, features)), dim=1)
+        data_input = torch.cat((data_input, torch.randn(batch, time + time_padding, features)), dim=0)
+
+        sequence_mask = torch.ones(batch, time)
+        sequence_mask = torch.cat((sequence_mask, torch.zeros(batch, time_padding)), dim=1)
+        sequence_mask = torch.cat((sequence_mask, torch.ones(batch, time + time_padding)), dim=0)
```

Comment:
This `cat` does not make sense to me. Why are you concatenating in the batch dimension? So the output tensor has shape [batch * 2, time + time_padding]. That looks wrong.

Reply:
in order to simulate having different seq lengths within one batch.. maybe you know a cleaner way to create this?

Reply:
But you have 2 * batch dim here? Already the shape looks wrong.

But if you just want to have random data, why not doing that more directly, using `numpy.random` or so?

E.g. see this function: https://github.com/rwth-i6/returnn/blob/3d148398e2f2177b896a03b9b6eafb1ad9863f16/returnn/tensor/utils.py#L35

Or this: https://github.com/rwth-i6/returnn/blob/3d148398e2f2177b896a03b9b6eafb1ad9863f16/tests/test_TFNetworkLayer.py#L34

Reply:
I ll check those